class-amp-instagram-embed.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+require_once( dirname( __FILE__ ) . '/class-amp-embed-handler.php' );
+
+// Much of this class is borrowed from Jetpack embeds
+class AMP_Instagram_Embed_Handler extends AMP_Embed_Handler {
+	const SHORT_URL_HOST = 'instagr.am';
+	const URL_PATTERN = '#http(s?)://(www\.)?instagr(\.am|am\.com)/p/([^/]*)#i';
```

Comment:
We can update the last selector (`([^/]*)`) to ignore `?` as well using `([^/?]+)` to select the full id.